docs/javascript/components_google_maps.md

```diff
@@ -0,0 +1,173 @@
+# Google Maps - JavaScript API
+
+The Google Maps component is used to show a map using the Google Maps API.
+
+## Example
+
+The component can be included directly as follows:
+
+```html
+<woltlab-core-google-maps
+    id="id"
+    class="googleMap"
+    api-key="your_api_key"
+></woltlab-core-google-maps>
+```
+
+Alternatively, the component can be included via a template that uses the API key from the configuration and also handles the user content:
+
+```smarty
+{include file='googleMapsElement' googleMapsElementID="id"}
+```
+
+## Parameters
+
+### `id`
+
+ID of the map instance.
+
+### `api-key`
+
+Google Maps API key.
+
+### `zoom`
+
+Defaults to `13`.
+
+Default zoom factor of the map.
+
+### `lat`
+
+Defaults to `0`.
+
+Latitude of the default map position.
+
+### `lng`
+
+Defaults to `0`.
+
+Longitude of the default map position.
+
+### `access-user-location`
+
+If set, the map will try to center based on the user's current position.
+
+## Map-related Functions
+
+### `addMarker`
+
+Adds a marker to the map.
+
+#### Example
+
+```html
+<script data-relocate="true">
+	require(['WoltLabSuite/Core/Component/GoogleMaps/Marker'], ({ addMarker }) => {
+		addMarker(document.getElementById('map_id'), 52.4505, 13.7546, 'Title', true);
```

Comment:
```suggestion
		void addMarker(document.getElementById('map_id'), 52.4505, 13.7546, 'Title', true);
```

Although not required in this context, it’s helpful to mark otherwise dangling promises to indicate that something is happening here. This is especially true because the previous API provided a synchronous call to add markers.